backend.native/tests/codegen/associatedObjects/associatedObjects1.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package codegen.associatedObjects.associatedObjects1
+
+import kotlin.test.*
+import kotlin.reflect.*
+
+@Test
+@UseExperimental(ExperimentalAssociatedObjects::class)
+fun testBasics1() {
+    assertSame(Bar, Foo::class.findAssociatedObject<Associated1>())
+    assertSame(Baz, Foo::class.findAssociatedObject<Associated2>())
+    assertSame(null, Foo::class.findAssociatedObject<Associated3>())
+
+    assertSame(null, Bar::class.findAssociatedObject<Associated1>())
+}
+
+@UseExperimental(ExperimentalAssociatedObjects::class)
+@AssociatedObjectKey
+@Retention(AnnotationRetention.BINARY)
+annotation class Associated1(val kClass: KClass<*>)
+
+@UseExperimental(ExperimentalAssociatedObjects::class)
+@AssociatedObjectKey
+@Retention(AnnotationRetention.BINARY)
+annotation class Associated2(val kClass: KClass<*>)
+
+@UseExperimental(ExperimentalAssociatedObjects::class)
+@AssociatedObjectKey
+@Retention(AnnotationRetention.BINARY)
+annotation class Associated3(val kClass: KClass<*>)
+
+@Associated1(Bar::class)
+@Associated2(Baz::class)
+class Foo
+
+object Bar
+object Baz
+
+@Test
+@UseExperimental(ExperimentalAssociatedObjects::class)
+fun testGlobalOptimizations1() {
+    val i1 = I1ImplHolder::class.findAssociatedObject<Associated1>()!! as I1
```

Comment:
Is `findAssociatedObject` right? Maybe `getAssociatedObject`?

Reply:
This variant was selected on design meeting.

Reply:
Shalln't this method take second type parameter of object type and be reified on that?

Reply:
See [above](https://github.com/JetBrains/kotlin-native/pull/2947#discussion_r287829580).